apps/wallet/src/ui/app/pages/home/transfer-coin/TransferCoinForm/StepTwo.tsx

```diff
@@ -42,21 +46,32 @@ function StepTwo({
         values: { amount, to },
     } = useFormikContext<FormValues>();
 
-    const intl = useIntl();
-
     const onClearRef = useRef(onClearSubmitError);
     onClearRef.current = onClearSubmitError;
 
     useEffect(() => {
         onClearRef.current();
     }, [amount, to]);
 
-    const totalAmount =
-        (GAS_SYMBOL === coinSymbol ? parseFloat(amount) : 0) +
-        DEFAULT_GAS_BUDGET_FOR_TRANSFER;
+    const [decimals] = useCoinDecimals(coinType);
+    const amountWithoutDecimals = useMemo(
+        () => new BigNumber(amount).shiftedBy(decimals).toString(),
```

Comment:
Not sure if we should force it here to integer something like `.dp()`

Reply:
It should already effectively be forced to be decimals because the validation on the input, but I'll add a check just to be safe.